src/main/java/se/iths/springbootgroupproject/configurations/SecurityConfig.java

```diff
@@ -0,0 +1,56 @@
+package se.iths.springbootgroupproject.configurations;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchy;
+import org.springframework.security.access.hierarchicalroles.RoleHierarchyImpl;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler;
+import org.springframework.web.client.RestClient;
+import se.iths.springbootgroupproject.services.github.GithubOAuth2UserService;
+
+@Configuration
+public class SecurityConfig {
+
+    private final GithubOAuth2UserService githubOAuth2UserService;
+
+    public SecurityConfig(GithubOAuth2UserService githubOAuth2UserService) {
+        this.githubOAuth2UserService = githubOAuth2UserService;
+    }
+
+    @Bean
+    SecurityFilterChain web(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests(requests -> requests
+                        .requestMatchers("/web/welcome", "/login", "/oauth/**", "/logout", "/error**","/static/**").permitAll()
+                        .requestMatchers("/web/users/profiles/demo").authenticated()
```

Comment:
You missed adding "/web/messages" to the authenticated requestMatchers.
```suggestion
                        .requestMatchers("/web/users/profiles/demo","/web/messages").authenticated()
```